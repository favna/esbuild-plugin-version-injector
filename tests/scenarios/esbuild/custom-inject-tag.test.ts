import esbuild, { type BuildOptions } from 'esbuild';
import type { PluginOptions } from '../../../dist';
import { assertFileContent, buildAbsolutePath, createEsbuildConfig } from '../../common/util';

function createConfig(pluginOptions?: PluginOptions): BuildOptions {
  return createEsbuildConfig(
    {
      format: 'esm',
      outExtension: {
        '.js': '.mjs'
      },
      entryPoints: [buildAbsolutePath('../fixtures/build-in/custom-inject-tag.mts')]
    },
    pluginOptions
  );
}

describe('Custom Version Inject Tag', () => {
  test('GIVEN custom tag THEN injects version', async () => {
    const config = createConfig({
      injectTag: '[VersionInject][/VersionInject]'
    });

    await esbuild.build(config);

    await assertFileContent('../fixtures/build-out/esbuild/custom-inject-tag.mjs');
  });
});
