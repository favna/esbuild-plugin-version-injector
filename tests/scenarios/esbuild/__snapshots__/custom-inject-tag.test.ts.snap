// Vitest Snapshot v1

exports[`Custom Version Inject Tag > GIVEN custom tag THEN injects version 1`] = `
"// tests/fixtures/build-in/custom-inject-tag.mts
var version = \\"1.0.2\\";
export {
  version
};
"
`;
